lib/events/events.proto

```diff
@@ -937,6 +937,75 @@ message KubeRequest {
     int32 ResponseCode = 11 [ (gogoproto.jsontag) = "response_code" ];
 }
 
+// AppSessionStart is emitted when a user is issued an application certificate.
+message AppSessionStart {
+    // Metadata is a common event metadata
+    Metadata Metadata = 1
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // User is a common user event metadata
+    UserMetadata User = 2
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // SessionMetadata is a common event session metadata
+    SessionMetadata Session = 3
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // ServerMetadata is a common server metadata
+    ServerMetadata Server = 4
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // ConnectionMetadata holds information about the connection
+    ConnectionMetadata Connection = 5
+        [ (gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = "" ];
+
+    // PublicAddr is the public address of the application being requested.
+    string PublicAddr = 7 [ (gogoproto.jsontag) = "public_addr" ];
```

Comment:
Also, do you think it would be useful to have additional information about the application in addition to the public address? For example, the "app name" from the config so it's easy to identify which app is being proxied.